File: chat-model-experiments/openai-features/response_api.js
import OpenAI from "openai";
import "dotenv/config";

// using openAI response API to manege conversation.
//  just send the previous response id. for each request. it will remember the context.
//  But charges are applied in same way as chat completion API. (using messages array)

const openai = new OpenAI({
  apiKey: process.env.OPENAI_API_KEY,
});

const response = await openai.responses.create({
  model: "gpt-4o-mini",
  input: "i have 5 pets",
  store: true,
});

console.log(response.output_text);

const secondResponse = await openai.responses.create({
  model: "gpt-4o-mini",
  previous_response_id: response.id,
  input: [{ role: "user", content: "They are very funny." }],
  store: true,
});

console.log(secondResponse.output_text);

const thirdResponse = await openai.responses.create({
  model: "gpt-4o-mini",
  previous_response_id: secondResponse.id,
  input: [{ role: "user", content: "How many pets do i have?" }],
  store: true,
});

console.log(thirdResponse.output_text);

File: chat-model-experiments/backend/services/response_api.py
from openai import OpenAI
import os

#  history taking model

API_KEY = os.environ.get("OPENAI_API_KEY")

client = OpenAI(
    api_key=API_KEY
)

def talk_to_llm(previous_response_id=None, input_text="" , system_prompt=""):

    response = client.responses.create(
        model="gpt-4.1",
        previous_response_id=previous_response_id,
        input=[{"role":"system", "content": system_prompt},{"role": "user", "content": input_text}],
        max_output_tokens=100
    )

    return response

